src/plugins/charts/common/palette.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ExpressionFunctionDefinition } from 'src/plugins/expressions/common';
+import { i18n } from '@kbn/i18n';
+
+export interface CustomPaletteArguments {
+  color?: string[];
+  gradient: boolean;
+  reverse?: boolean;
+}
+
+export interface CustomPaletteState {
+  colors: string[];
+  gradient: boolean;
+}
+
+export interface SystemPaletteArguments {
+  name: string;
+}
+
+export interface PaletteOutput<T = unknown> {
+  type: 'palette';
+  name: string;
+  params?: T;
+}
+export const defaultCustomColors = [
```

Comment:
```suggestion
// This set of defaults originated in Canvas, which, at present, is the primary 
// consumer of this function.  Changing this default requires a change in Canvas
// logic, which would likely be a breaking change in 7.x.
export const defaultCustomColors = [
```

Reply:
Ah, forgot about that, thanks for the great suggestion!